core/templates/services/guppy-initialization.service.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for initializing guppy instances.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+class GuppyObject {
+  divId = null;
+  guppyInstance = null;
+  constructor(divId: string, guppyInstance: Guppy) {
+    this.divId = divId;
+    this.guppyInstance = guppyInstance;
+  }
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GuppyInitializationService {
+  private guppyInstances: Array<GuppyObject> = [];
+
+  init(guppyDivClassName: string): void {
+    var guppyDivs = document.querySelectorAll('.' + guppyDivClassName);
+    var divId, guppyInstance;
+    for (var i = 0; i < guppyDivs.length; i++) {
+      divId = 'guppy_' + Math.floor(Math.random() * 100000000);
+      // Dynamically assigns a unique id to the guppy div.
+      guppyDivs[i].setAttribute('id', divId);
+      // Create a new guppy instance for that div.
+      guppyInstance = new Guppy(divId, {});
+      this.guppyInstances.push(new GuppyObject(divId, guppyInstance));
+    }
+  }
```

Comment:
```suggestion
    for (let guppyDiv of guppyDivs) {
      divId = 'guppy_' + Math.floor(Math.random() * 100000000);
      // Dynamically assigns a unique id to the guppy div.
      guppyDiv.setAttribute('id', divId);
      // Create a new guppy instance for that div.
      guppyInstance = new Guppy(divId, {});
      this.guppyInstances.push(new GuppyObject(divId, guppyInstance));
    }
  }
```

Reply:
I'd done this before but this doesn't work. Since `document.querySelectorAll` returns a `NodeList`, you can't iterate using `of`.

Reply:
Right, NodeList doesn't inherit from array. So it is either forEach or the normal for loop. I think we can go with the normal for loop.